esx_service/utils/auth_api.py

```diff
@@ -432,13 +454,24 @@ def _tenant_vm_add(name, vm_list):
         error_info = error_code.generate_error_info(ErrorCode.VM_ALREADY_IN_TENANT, vm_list, name)
         return error_info
 
+    error_info, tenant_list = get_tenant_list_from_db()
```

Comment:
Same as above - this logic should not be triggered if vm-list is None. For this method, however, I think we should return right away if vm-list is None at the very first beginning (after line 434) to avoid unnecessary cost, because the whole purpose of this method is to add VMs to a tenant.